src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Automation/Peers/MenuItemAutomationPeer.cs

```diff
@@ -118,13 +126,26 @@ override protected int GetSizeOfSetCore()
         override protected int GetPositionInSetCore()
         {
             int positionInSet = base.GetPositionInSetCore();
-
+            
             if (positionInSet == AutomationProperties.AutomationPositionInSetDefault)
             {
-                MenuItem owner = (MenuItem)Owner;
+                MenuItem owner = (MenuItem)Owner;
                 ItemsControl parent = ItemsControl.ItemsControlFromItemContainer(owner);
-                positionInSet = ItemAutomationPeer.GetPositionInSetFromItemsControl(parent, owner);
-            }
+
+                positionInSet = ItemAutomationPeer.GetPositionInSetFromItemsControl(parent, owner);
+                
+                foreach (var item in parent.Items)
+                {
+                    if (item == owner)
+                    {
+                        break;

```

Comment:
Is it `continue` ?

Reply:
no, you don't want to skip the owner, you want to stop at the owner (what you actually want to do is visit all items before the owner, so seeing the owner is your signal to end the loop)